VoodooI2CHID/VoodooI2CPrecisionTouchpadHIDEventDriver.cpp

```diff
@@ -12,8 +12,25 @@
 OSDefineMetaClassAndStructors(VoodooI2CPrecisionTouchpadHIDEventDriver, VoodooI2CMultitouchHIDEventDriver);
 
 void VoodooI2CPrecisionTouchpadHIDEventDriver::enterPrecisionTouchpadMode() {
-    digitiser.input_mode->setValue(INPUT_MODE_TOUCHPAD);
+    if (version_major > CATALINA_MAJOR_VERSION) {
+        // Update value from hardware so we can rewrite mode when waking from sleep
+        digitiser.input_mode->getValue(kIOHIDValueOptionsUpdateElementValues);
```

Comment:
If the value in the input mode element is the same as what we try to set, setValue immediately returns without updating the value in hardware. This updates the element from hardware so that it doesn't immediately return when trying to set a new one.

https://github.com/apple-opensource/IOHIDFamily/blob/master/IOHIDFamily/IOHIDElementPrivate.cpp#L1855-L1861

It will always set the value the first time, which is why we only see this when waking from sleep.